lib/datadog/kit/events.rb

```diff
@@ -7,41 +7,62 @@ module Datadog
+
+      # Attach login success event information to the trace
+      #
+      # This method is experimental and may change in the future.
       #
       # @param trace [TraceOperation] Trace to attach data to.
-      # @param success [bool] Whether login operation was successful or not.
       # @param user [Hash<Symbol, String>] User information to pass to
       #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
       # @param others [Hash<String || Symbol, String>] Additional free-form
       #   event information to attach to the trace.
-      def self.track_login(trace, success:, user:, **others)
-        subtag = success ? 'success' : 'failure'
-        event = "users.login.#{subtag}"
-
-        track(event, trace, **others)
+      def self.track_login_success(trace, user:, **others)
+        track(:appsec, APPSEC_LOGIN_SUCCESS_EVENT, trace, **others)
 
         user_options = user.dup
         user_id = user.delete(:id)
 
         raise ArgumentError, 'missing required key: :user => { :id }' if user_id.nil?
 
-        if success
-          Kit::Identity.set_user(trace, id: user_id, **user_options)
-        else
-          trace.set_tag("appsec.events.users.login.#{subtag}.usr.id", user_id)
-        end
+        Kit::Identity.set_user(trace, id: user_id, **user_options)
+      end
+
+      # Attach login failure event information to the trace
+      #
+      # This method is experimental and may change in the future.
+      #
+      # @param trace [TraceOperation] Trace to attach data to.
+      # @param user_id [String] User id that attempted login
+      #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
+      # @param user_exists [bool] Whether the user id that did a login attempt exists.
+      # @param others [Hash<String || Symbol, String>] Additional free-form
+      #   event information to attach to the trace.
+      def self.track_login_failure(trace, user_id:, user_exists:, **others)
+        track(:appsec, APPSEC_LOGIN_FAILURE_EVENT, trace, **others)
+
+        raise ArgumentError, 'missing required key: :user => { :id }' if user_id.nil?
+
+        trace.set_tag('appsec.events.users.login.failure.usr.id', user_id)
+        trace.set_tag('appsec.events.users.login.failure.usr.exists', user_exists)
       end
 
       # Attach custom event information to the trace
       #
+      # This method is experimental and may change in the future.
+      #
+      # @param namespace [Symbol] Mandatory. Event namespace. Only :appsec is supported.
       # @param event [String] Mandatory. Event code.
       # @param trace [TraceOperation] Trace to attach data to.
       # @param others [Hash<Symbol, String>] Additional free-form
       #   event information to attach to the trace. Key must not
       #   be :track.
-      def self.track(event, trace, **others)
-        trace.set_tag("appsec.events.#{event}.track", 'true')
+      def self.track(namespace, event, trace, **others)
```

Comment:
Since it's not entirely clear how this is going to be used going forward (or even beyond appsec), have you considered making this method private for now, e.g. making it so it can only be called by other methods in this file?

Reply:
That is a very interesting point. If this API will be available for users and we later want to change it it would be hard to do. 

I agree with the sentiment of making it private until we have more knowledge of how `track` is going to be use

Reply:
It's part of the public API of the event RFC, so we can't go private here. I'm thinking about moving these public things either to the `Datadog::AppSec` namespace (that would mean introducing something like `Datadog::AppSec::Kit`) or to a `Datadog::Kit::AppSec` namespace.

Reply:
I will try a namespace move in a subsequent PR so that we can merge this one.